raco/backends/myria/myria.py

```diff
@@ -2437,6 +2448,11 @@ def compile_to_json(raw_query, logical_plan, physical_plan,
     assert isinstance(physical_plan, subplan_ops), \
         'Physical plan must be a subplan operator, not {}'.format(type(physical_plan))  # noqa
 
+    if('OrderBy' in str(logical_plan)
```

Comment:
I don't see how directly storing an `OrderBy` result makes sense. There is no ordering guarantee from a subsequent `scan()` on that stored relation (until we implement ordering metadata in the catalog and automatic index creation/`ORDER BY` SQL generation for the scan).

As for your bag comprehension example with `ORDER BY`, I think the *cough* ordering of the `ORDER BY` clause is causing confusion. I think you are inferring from the fact that the `ORDER BY` clause precedes the `EMIT` clause that the `OrderBy` operator should precede the `Apply` operator. I don't think that's the case in SQL, and I don't see why it should be the case here. I don't see why we shouldn't place the `ORDER BY` clause last in the bag comprehension syntax, making it clear that the `OrderBy` operator is the root of the operator tree (well, assuming that `LIMIT` isn't present, which it must be for now).

A sufficiently smart optimizer might push the `OrderBy` operator below an aggregate so it could use a sorted aggregate implementation, but that is orthogonal to this change (and again would require us to propagate ordering information between operators, starting at scans).